### DISCUSSION TITLE
Feature request: Time shift rec/play

### DISCUSSION THREAD

**Discussion by FredHucht** [AlexandreRouma/SDRPlusPlus#965] [Ideas]
Hi and thanks for this excellent software!

I have a feature request that I didn't find in the discussions, which might be quite easy to implement: 

If SDR++ would save the whole baseband data into a ring buffer of the size of, say, the waterfall, one could implement "Time Shift rec/play": By clicking into the waterfall at a certain time in the past, the decoding and playback would happen at that time instead at the actual time. The playback time could be marked with a horizontal line. With such a feature, it would be easy to listen to short radio comms as in air radio. Furthermore, one could "see the future" and jump to signals in advance.

The memory requirements should be ~500MB/min at 2.048 MHz bandwidth, so several minutes should fit into memory.

What do you think?

Best, Fred

**Comment by rob-fi** [comment]
Found this discussion when searching to see if there was a solution for SDR++. There was an old plugin for SDR# about 10 years ago that did just this and it was so fun especially on HF!
